MonashNovaRover/nova
# Create useGenericStore

## Description


Created useGenericStore to replace useLocalStorage. Basically a common and easy way to create redux stores which can then be easily used in a `[value, setValue]` way.

See [the doc I wrote](https://github.com/MonashNovaRover/nova/blob/7b64798df4f45cf712c7a5799134ce9a8a56debf/src/ros/nova-gui/docs/generic_store.md) for more detail.

fixes: #9 and helps #5 

Used `currentSite` as a way to test, will implement the SiteSelectWidget changes in a future PR.




## Changelogs


- create `createGenericStore`
- create `useGenericStore`
- when adding a new store default is to persist and tab sync -> the store must be manually blacklisted from both services if this functionality is not desired.

## Screenshots

[Screencast from 2024-09-30 09-12-49.webm](https://github.com/user-attachments/assets/593edc30-20c2-4cec-a502-0dfa14cbf6d9)


## Steps to Test



run the gui and check out the site selector and bottom counters. They both use `useGenericStore` and should persist and tab sync.




## Memes
![useGenericStore](https://github.com/user-attachments/assets/bd1d21f6-efff-461b-b83f-af64cb9d62bb)
